File: Java/Ema/Core/src/main/java/com/refinitiv/ema/access/OmmJson.java
///*|-----------------------------------------------------------------------------
// *|            This source code is provided under the Apache 2.0 license
// *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
// *|                See the project's LICENSE.md for details.
// *|           Copyright (C) 2024 LSEG. All rights reserved.
///*|-----------------------------------------------------------------------------

package com.refinitiv.ema.access;

import java.nio.ByteBuffer;

/**
 * OmmJson represents JSON data format in Omm.
 *
 * <p>Method {@link #string()} can have performance decrease, since it will trigger garbage collection.</p>
 * <p>Objects of this class are intended to be short lived or rather transitional.<br>
 * <br>This class is designed to efficiently perform setting and extracting of JSON and its content.<br>
 * <br>Objects of this class are not cache-able.</p>
 *
 * @see Data
 */
public interface OmmJson extends ComplexType
{
    /**
     * Returns contained JSON buffer.
     *
     * @return ByteBuffer containing the JSON data
     */
    ByteBuffer buffer();

    /**
     * Returns a string representation of the class instance.
     *
     * @return string representation of the class instance
     */
    String string();

    /**
     * Clears the OmmJson.
     * Invoking clear() method clears all the values and resets all the defaults.
     *
     * @return reference to this object
     */
    OmmJson clear();

    /**
     * Specifies Set.
     *
     * @param value specifies JSON data using String
     * @return reference to this object
     */
    OmmJson string(String value);

    /**
     * Specifies Set.
     *
     * @param value specifies JSON data using ByteBuffer
     * @return reference to this object
     */
    OmmJson buffer(ByteBuffer value);
}

File: CSharp/Ema/Src/Core/Access/OmmJson.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2025 LSEG. All rights reserved.
 *|-----------------------------------------------------------------------------
 */

namespace LSEG.Ema.Access
{
    /// <summary>
    /// OmmJson represents JSON data format in Omm.
    /// </summary>
    /// <remarks>
    /// Method <see cref="StructuredTextData.GetString"/> can have performance decrease, since it will trigger garbage collection.<br/>
    /// Objects of this class are intended to be short lived or rather transitional.<br/>
    /// This class is designed to efficiently perform setting and extracting of JSON and its content.<br/>
    /// Objects of this class are not cache-able.<br/>
    /// </remarks>
    public sealed class OmmJson : StructuredTextData
    {
        /// <summary>
        /// Constructor for OmmJson class
        /// </summary>
        public OmmJson()
        {
            ReturnToPoolInternal = () => m_objectManager.ReturnToPool(this);
            m_dataType = Access.DataType.DataTypes.JSON;
        }

        /// <summary>
        /// Clears the OmmJson instance.
        /// </summary>
        /// <returns>Reference to current <see cref="OmmJson"/> object.</returns>
        public OmmJson Clear()
        {
            Clear_All();
            return this;
        }

        /// <summary>
        /// Specifies Set.
        /// </summary>
        /// <param name="value">specifies JSON data using string</param>
        /// <returns>Reference to current <see cref="OmmJson"/> object.</returns>
        public new OmmJson SetString(string value)
        {
            base.SetString(value);
            return this;
        }

        /// <summary>
        /// Encodes specified EmaBuffer containing JSON data.
        /// </summary>
        /// <param name="value"> specifies JSON data using <see cref="EmaBuffer"/></param>
        /// <returns>Reference to current <see cref="OmmJson"/> object.</returns>
        public new OmmJson SetBuffer(EmaBuffer value)
        {
            base.SetBuffer(value);
            return this;
        }
    }
}